util/bitstream.go

```diff
@@ -275,3 +278,24 @@ func (bs *BitStream) ReadTwoBitField(numFields int) ([]byte, error) {
 
 	return result, nil
 }
+
+func (bs *BitStream) ReadByteSize(size int, err error) (byte, error) {
```

Comment:
Is there a reason to use `ReadByteSize(N, err)` rather than just `ReadByteN()`? And why pass an error in just to pass it back out without doing anything? (That might be the reason for `ReadByteSize()`)

Reply:
I initially used this function to try to clean up the setting of all the segments. However, I ended up removing this function now and just using the bitstream functions directly.